source/opt/folding_rules.cpp

```diff
@@ -2819,6 +2819,249 @@ FoldingRule RedundantSUMod() {
   };
 }
 
+// Helper for zipping together integer constants.
+// The input types must be
+// Integers or Vectors of Integers.
```

Comment:
Comment should be updated:

Utility function for applying `callback` to `input1` and `input2`. If they are vectors it applies element wise. The constants `input1` and `input2` must be integers or a vector of integers.